src/EFCore.PG/Extensions/NpgsqlNetworkExtensions.cs

```diff
@@ -0,0 +1,1006 @@
+        public static bool GreaterThan([CanBeNull] this DbFunctions _, PhysicalAddress macaddr, PhysicalAddress other) => throw new ClientEvaluationNotSupportedException();
+
+        /// <summary>
+        /// Determines whether an <see cref="IPAddress"/> is not equal to another <see cref="IPAddress"/>.
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="inet">The left-hand inet.</param>
+        /// <param name="other">The right-hand inet.</param>
+        /// <returns>
+        /// True if the <see cref="IPAddress"/> is not equal to the other <see cref="IPAddress"/>; otherwise, false.
+        /// </returns>
+        /// <exception cref="ClientEvaluationNotSupportedException">
+        /// This method is only intended for use via SQL translation as part of an EF Core LINQ query.
+        /// </exception>
+        public static bool NotEqual([CanBeNull] this DbFunctions _, IPAddress inet, IPAddress other) => throw new ClientEvaluationNotSupportedException();
+
+        /// <summary>
+        /// Determines whether an (IPAddress Address, int Subnet) is not equal to another (IPAddress Address, int Subnet).
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="cidr">The left-hand cidr.</param>
+        /// <param name="other">The right-hand cidr.</param>
+        /// <returns>
+        /// True if the IPAddress Address, int Subnet) is not equal to the other (IPAddress Address, int Subnet); otherwise, false.
+        /// </returns>
+        /// <exception cref="ClientEvaluationNotSupportedException">
+        /// This method is only intended for use via SQL translation as part of an EF Core LINQ query.
+        /// </exception>
+        public static bool NotEqual([CanBeNull] this DbFunctions _, (IPAddress Address, int Subnet) cidr, (IPAddress Address, int Subnet) other) => throw new ClientEvaluationNotSupportedException();
+
+        /// <summary>
+        /// Determines whether an <see cref="PhysicalAddress"/> is not equal to another <see cref="PhysicalAddress"/>.
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="macaddr">The left-hand macaddr.</param>
+        /// <param name="other">The right-hand macaddr.</param>
+        /// <returns>
+        /// True if the <see cref="PhysicalAddress"/> is not equal to the other <see cref="PhysicalAddress"/>; otherwise, false.
+        /// </returns>
+        /// <exception cref="ClientEvaluationNotSupportedException">
+        /// This method is only intended for use via SQL translation as part of an EF Core LINQ query.
+        /// </exception>
+        public static bool NotEqual([CanBeNull] this DbFunctions _, PhysicalAddress macaddr, PhysicalAddress other) => throw new ClientEvaluationNotSupportedException();
+
+        #endregion
+
+        #region ContainmentOperators
```

Comment:
Aren't we missing the check of containment of `inet` in a `cidr`, which seems to be the most relevant one? Or does PostgreSQL not support this somehow? Same for some of the other operations here.

Reply:
The implicit casting between `inet` and `cidr` makes it hard to decide when to stop including overloads, but I think you're right on the containment checks being important. 

I've added a few more in the next commit. Happy to add more, just let me know if there are others you want to see included.